custom_components/tuya_local/devices/thermoval_tvt40_thermostat.yaml

```diff
@@ -0,0 +1,315 @@
+name: Thermoval WiFi Thermostat
+products:
+  - id: jIRwsT5yLPJO41Po
+    name: TVT 40
+primary_entity:
+  entity: climate
+  translation_key: thermostat
+  dps:
+    - id: 2
+      name: temperature
+      type: integer
+      range:
+        min: 50
+        max: 400
+      mapping:
+        - scale: 10
+          step: 5
+    - id: 3
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: Smart
+          value: program
+          icon: "mdi:home-thermometer"
+        - dps_val: Anti_frozen
+          value: away
+          icon: "mdi:snowflake-melt"
+        - dps_val: Manual
+          value: manual
+          icon: "mdi:cursor-pointer"
+    - id: 4
+      name: fault
+      type: integer
+    - id: 5
+      name: room_temperature

```

Comment:
You may want to make this conditionally redirect to floor_temperature when sensor_select is "0", the config will be more complex, but the UI might make more sense in terms of offset between the set temperature and current temperature.  If you do this, then keep the room temperature separate sensor, so that it is always available regardless of the sensor select setting.